presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2005,6 +2005,27 @@ public PartitionStatistics getStatisticsUpdate()
             return statisticsUpdate;
         }
 
+        public Table getAugmentTableForInTransactionRead()
+        {
+            if (!this.currentLocation.isPresent()) {
+                // partitioned table
+                return this.table;
+            }
+
+            // For unpartitioned table, this method augments the location field of the table
+            // to the staging location.
+            // This way, if the table is accessed in an ongoing transaction, staged data
+            // can be found and accessed.
+            Table table = this.table;
+            String currentLocation = this.currentLocation.get().toString();
+            if (!currentLocation.equals(table.getStorage().getLocation())) {
+                table = Table.builder(table)
+                        .withStorage(storage -> storage.setLocation(currentLocation))
```

Comment:
Is this only used for reads?  Would this have any side effects for writing the table?
e.g. I create a table and then insert into it, will the location be wrong?

Reply:
Currently we don't allow multiple inserts into a same table / partition within the same transaction.